src/Rotations.jl

```diff
@@ -0,0 +1,148 @@
+@doc doc"""
+    Rotations{N} <: Manifold
+
+Special orthogonal manifold $\mathrm{SO}(n)$ represented by $n\times n$
+real-valued orthogonal matrices with determinant $+1$.
+
+# Constructor
+
+    Rotations(n, retract! = rectract_qr!)
+
+generates the $\mathrm{SO}(n) \subset \mathbb R^{n\times n}$
+"""
+struct Rotations{N, TRetr<:Function} <: Manifold
+    retract!::TRetr
+end
+Rotations(n::Int, retract!::TRetr = retract_qr!) where TRetr = Rotations{n,TRetr}(retract!)
```

Comment:
ah, also neat. I am currently putting the choice for the retraction into the problem (but we don't have that here of course) in the next version of Manopt.jl. This way introduces an implicit dispatch for the default retraction. Nice idea.

Reply:
Oh and what about the default inverse retraction (first/second/higher order approximation of log)?

Reply:
I skipped inverse retractions for now, I'll add them soon.

Reply:
Huh. I don't think I like this. `Rotations` will ultimately be decorated with a `SpecialOrthogonal` group structure and could then be decorated with a different metric and/or combined via a semidirect product to get a `SpecialEuclidean` element. It seems odd to have the base rotation manifold specialized on the retraction, which in many cases may not be used. I get that the desired behavior is to specify a default retraction for an optimization problem, but I wonder if there's a better way to do it that doesn't bake it into the manifold itself.

Since what differentiates these retractions is their order, I wonder if a good way to go would be to let `retract!` take an order parameter, which it would round up. So if a 2nd-order approximation is requested it would retract using the 2nd order approximation if available and perform the `exp!` map if not. Then you could do have something like `retraction(M,order=1)` that would return a suitable retraction of at least that order for your manifold.

Reply:
Specifying retraction order (or method) by an additional parameter is a good idea, but I'd prefer to have it resolved by dispatch instead. For example 
```julia
retract!(M::Manifold, y, x, v, method) = exp!(M, y, x, v)

retract!(M::Rotations, y, x, v, ::Val{:polar}) = ...
```

Reply:
I like that the mabnifold provides a default retration, it can still be overwritten by a decorator, why not?
but, hm, yeah a dispatch by value is fine, since in the end that also provides a proper default.

Reply:
OK, done. I've decided to make an abstract type, it will be better once we get to product manifolds.